docs/resources/Poll.md

```diff
@@ -0,0 +1,138 @@
+# Poll Resource
+
+A poll is... well... a poll! It holds information about a poll!
+
+![Example message containing a poll](example-poll.png)
+
+### Poll Object
+
+The poll object has a lot of levels and nested structures. It was also designed
+to support future extensibility, so some fields may appear to be more complex than
+necessary.
+
+###### Poll Object Structure
+
+| Field             | Type                                                                                                | Description                                                     |
+|-------------------|-----------------------------------------------------------------------------------------------------|-----------------------------------------------------------------|
+| question          | [Poll Media Object](#DOCS_RESOURCES_POLL/poll-media-object-poll-media-object-structure)             | The question of the poll. Only `text` is supported.             |
+| answers           | List of [Poll Answer Objects](#DOCS_RESOURCES_POLL/poll-answer-object-poll-answer-object-structure) | Each of the answers available in the poll.                      |
+| expiry            | IS08601 timestamp                                                                                   | The time when the poll ends.                                    |
```

Comment:
would be nice if expiry was nullable to support infinite polls - many servers already use reactions for infinite polls and users would like to be able to do that with native polls also. maybe this could be changed in the future but making a non-nullable field nullable would be a breaking change

Reply:
Done. I think regardless of us doing it this is a good change for API compat.